File: java-client/src/main/java/com/bolour/boardgame/client/domain/PlayPiece.java
/*
 * Copyright 2017 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PlayPiece {
    public final Piece piece;
    public final Point point;
    public final boolean moved;

    @JsonCreator
    public PlayPiece(
      @JsonProperty("piece") Piece piece,
      @JsonProperty("point") Point point,
      @JsonProperty("moved") boolean moved
    ) {
        this.piece = piece;
        this.point = point;
        this.moved = moved;
    }
}

File: scala-server/boardgamecommon/src/main/scala/com/bolour/boardgame/scala/common/domain/PlayPiece.scala
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.boardgame.scala.common.domain

import com.bolour.plane.scala.domain.Point

case class PlayPiece(piece: Piece, point: Point, moved: Boolean) {
  def piecePoint = PiecePoint(piece, point)
}

// Companion object renamed to avoid akka http json format issue.

object PlayPieceObj {
  type PlayPieces = List[PlayPiece]
  def playPiecesToWord(playPieces: List[PlayPiece]): String =
    (playPieces map { _.piece.value }).mkString
}
